--- original/ScoreToHealth/data/score_to_health/functions/check.mcfunction
+++ modified/ScoreToHealth/data/score_to_health/functions/check.mcfunction
@@ -15,3 +15,4 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute store result score #__ ScoreToHealth run data get entity @s Health 100
-scoreboard players operation #__ ScoreToHealth -= #_ ScoreToHealth
+execute store result score $_ ScoreToHealth run attribute @s generic.max_health get 100
+function api:data_get/health
+execute store result score $__ ScoreToHealth run data get storage api: Health 100
+scoreboard players operation $__ ScoreToHealth -= $_ ScoreToHealth
@@ -19 +20 @@
-execute if score #__ ScoreToHealth matches ..0 unless entity @s[tag=ScoreToHealth.AntiGlitch.UnsafeTick] run function score_to_health:modify
+execute if score $__ ScoreToHealth matches ..0 unless entity @s[tag=ScoreToHealth.AntiGlitch.UnsafeTick] run function score_to_health:modify
@@ -21 +22 @@
-execute if score #__ ScoreToHealth matches 1.. run function score_to_health:anti_glitch/protect
+execute if score $__ ScoreToHealth matches 1.. run function score_to_health:anti_glitch/protect
--- original/ScoreToHealth/data/score_to_health/functions/load.mcfunction
+++ modified/ScoreToHealth/data/score_to_health/functions/load.mcfunction
@@ -14,0 +15,5 @@
+#> ObjectDeclare
+# @within *
+#   score_to_health:**
+#   lib:score_to_health_wrapper/**
+#   lib:has_health_modify_score
@@ -15,0 +21,7 @@
+
+#> PrivateUse
+# @internal
+    #declare tag ScoreToHealth.AntiGlitch.UnsafeTick
+    #declare tag ScoreToHealth.AntiGlitch.Protected
+    #declare tag ScoreToHealth.Modified
+    #declare tag ScoreToHealth.Return
--- original/ScoreToHealth/data/score_to_health/functions/modify.mcfunction
+++ modified/ScoreToHealth/data/score_to_health/functions/modify.mcfunction
@@ -17,2 +17,2 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-scoreboard players operation @s ScoreToHealth < #_ ScoreToHealth
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+scoreboard players operation @s ScoreToHealth < $_ ScoreToHealth
@@ -21,3 +21,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-16 "ScoreToHealth" -419.4304 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-16 "ScoreToHealth" 419.4304 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-16 "ScoreToHealth" -419.4304 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-16 "ScoreToHealth" 419.4304 add
@@ -25,3 +25,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-15 "ScoreToHealth" -209.7152 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-15 "ScoreToHealth" 209.7152 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-15 "ScoreToHealth" -209.7152 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-15 "ScoreToHealth" 209.7152 add
@@ -29,3 +29,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-14 "ScoreToHealth" -104.8576 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-14 "ScoreToHealth" 104.8576 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-14 "ScoreToHealth" -104.8576 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-14 "ScoreToHealth" 104.8576 add
@@ -33,3 +33,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-13 "ScoreToHealth" -52.4288 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-13 "ScoreToHealth" 52.4288 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-13 "ScoreToHealth" -52.4288 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-13 "ScoreToHealth" 52.4288 add
@@ -37,3 +37,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-12 "ScoreToHealth" -26.2144 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-12 "ScoreToHealth" 26.2144 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-12 "ScoreToHealth" -26.2144 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-12 "ScoreToHealth" 26.2144 add
@@ -41,3 +41,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-11 "ScoreToHealth" -13.1072 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-11 "ScoreToHealth" 13.1072 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-11 "ScoreToHealth" -13.1072 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-11 "ScoreToHealth" 13.1072 add
@@ -45,3 +45,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-10 "ScoreToHealth" -6.5536 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-10 "ScoreToHealth" 6.5536 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-10 "ScoreToHealth" -6.5536 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-10 "ScoreToHealth" 6.5536 add
@@ -49,3 +49,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0F "ScoreToHealth" -3.2768 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0F "ScoreToHealth" 3.2768 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0F "ScoreToHealth" -3.2768 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0F "ScoreToHealth" 3.2768 add
@@ -53,3 +53,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0E "ScoreToHealth" -1.6384 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0E "ScoreToHealth" 1.6384 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0E "ScoreToHealth" -1.6384 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0E "ScoreToHealth" 1.6384 add
@@ -57,3 +57,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0D "ScoreToHealth" -0.8192 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0D "ScoreToHealth" 0.8192 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0D "ScoreToHealth" -0.8192 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0D "ScoreToHealth" 0.8192 add
@@ -61,3 +61,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0C "ScoreToHealth" -0.4096 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0C "ScoreToHealth" 0.4096 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0C "ScoreToHealth" -0.4096 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0C "ScoreToHealth" 0.4096 add
@@ -65,3 +65,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0B "ScoreToHealth" -0.2048 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0B "ScoreToHealth" 0.2048 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0B "ScoreToHealth" -0.2048 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0B "ScoreToHealth" 0.2048 add
@@ -69,3 +69,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0A "ScoreToHealth" -0.1024 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0A "ScoreToHealth" 0.1024 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0A "ScoreToHealth" -0.1024 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-0A "ScoreToHealth" 0.1024 add
@@ -73,3 +73,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-09 "ScoreToHealth" -0.0512 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-09 "ScoreToHealth" 0.0512 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-09 "ScoreToHealth" -0.0512 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-09 "ScoreToHealth" 0.0512 add
@@ -77,3 +77,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-08 "ScoreToHealth" -0.0256 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-08 "ScoreToHealth" 0.0256 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-08 "ScoreToHealth" -0.0256 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-08 "ScoreToHealth" 0.0256 add
@@ -81,3 +81,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-07 "ScoreToHealth" -0.0128 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-07 "ScoreToHealth" 0.0128 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-07 "ScoreToHealth" -0.0128 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-07 "ScoreToHealth" 0.0128 add
@@ -85,3 +85,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-06 "ScoreToHealth" -0.0064 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-06 "ScoreToHealth" 0.0064 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-06 "ScoreToHealth" -0.0064 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-06 "ScoreToHealth" 0.0064 add
@@ -89,3 +89,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-05 "ScoreToHealth" -0.0032 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-05 "ScoreToHealth" 0.0032 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-05 "ScoreToHealth" -0.0032 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-05 "ScoreToHealth" 0.0032 add
@@ -93,3 +93,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-04 "ScoreToHealth" -0.0016 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-04 "ScoreToHealth" 0.0016 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-04 "ScoreToHealth" -0.0016 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-04 "ScoreToHealth" 0.0016 add
@@ -97,3 +97,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-03 "ScoreToHealth" -0.0008 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-03 "ScoreToHealth" 0.0008 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-03 "ScoreToHealth" -0.0008 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-03 "ScoreToHealth" 0.0008 add
@@ -101,3 +101,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-02 "ScoreToHealth" -0.0004 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-02 "ScoreToHealth" 0.0004 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-02 "ScoreToHealth" -0.0004 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-02 "ScoreToHealth" 0.0004 add
@@ -105,3 +105,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-01 "ScoreToHealth" -0.0002 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-01 "ScoreToHealth" 0.0002 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-01 "ScoreToHealth" -0.0002 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-01 "ScoreToHealth" 0.0002 add
@@ -109,3 +109,3 @@
-execute store result score #_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
-execute if score @s ScoreToHealth <= #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-00 "ScoreToHealth" -0.0001 add
-execute if score @s ScoreToHealth > #_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-00 "ScoreToHealth" 0.0001 add
+execute store result score $_ ScoreToHealth run attribute @s minecraft:generic.max_health get 100
+execute if score @s ScoreToHealth <= $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-00 "ScoreToHealth" -0.0001 add
+execute if score @s ScoreToHealth > $_ ScoreToHealth run attribute @s minecraft:generic.max_health modifier add bab7cdc2-fb6a-47f6-0001-00 "ScoreToHealth" 0.0001 add
--- original/ScoreToHealth/pack.mcmeta
+++ modified/ScoreToHealth/pack.mcmeta
@@ -3 +3 @@
-		"pack_format": 10,
+        "pack_format": 26,
